syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -210,6 +210,60 @@ def id(self):
                 self._id = syft.ID_PROVIDER.pop()
                 return self._id
 
+    @property
+    def gc(self):
+        if self.has_child():
+            return self.child.garbage_collect_data
+        else:
+
+            if not hasattr(self, "_gc"):
+                self._gc = True
+            if hasattr(self, "ptr") and self.ptr is not None:
+                self.child = self.ptr
+                self.child.garbage_collect_data = self._gc
+                self.garbage_collection = self._gc
+            return self._gc
+
+    @gc.setter
+    def gc(self, flag):
+        if self.has_child():
+            self.child.garbage_collect_data = flag
+        else:
+            if hasattr(self, "ptr") and self.ptr is not None:
+                self.child = self.ptr
+                self.child.garbage_collect_data = flag
+        self._gc = flag
+
+    @property
+    def disable_gc(self):
+        self.child.garbage_collect_data = False
+        if self.has_gc():
+            self.gc = False
+            self.garbage_collection = False
+        return self
+
+    def has_gc(self):
+        return hasattr(self, "gc")
+
+    @property
+    def garbage_collection(self):
```

Comment:
what is the difference between this and gc?

Reply:
It should be the same - I thought it might be nice to have a shorthand version. Open to comments there.

Reply:
My point for this is that repoducing the logic in 2 places is bad. If we change one of them and forget to change the other, that leads to unexpected behavior

Reply:
Nothing, gc just shorthand form. It was mentioned in the description https://github.com/OpenMined/PySyft/issues/2444

Reply:
go with Bobby's suggestion - he's usually right :)

Reply:
For clarification, I think we should keep both, just abstract them so that they call the same thing 🙂